instance/api/instance.py

```diff
@@ -47,7 +47,9 @@ def provision(self, request, pk=None):
         Start the (re-)provisioning of an instance
         """
         instance = self.get_object()
-        if instance.status not in (instance.EMPTY, instance.READY):
+        if instance.status not in (instance.EMPTY,
+                                   instance.READY,
+                                   instance.ERROR):
```

Comment:
IMHO it will provide undesired side effects, for example: if we do it and then proceed with the changes below to the UI, for a short period of time the user will see the provision button and then it will disappear, because the state NEW will only last for a few ms. Plus, it is one more state we will have to maintain.

I suggest to just set the default progress value to SUCCESS on the Server model, since the default status is NEW.